### PR TITLE
fix: forward context in idler reconciler

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: controllers/idler/idler_controller.go [hunk 1]
```diff
@@ -141,13 +141,18 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 				}
 
 				// Check if it belongs to a controller (Deployment, DeploymentConfig, etc) and scale it down to zero.
-				appType, appName, deletedByController, err := r.scaleControllerToZero(podLogger, pod.ObjectMeta)
+				appType, appName, deletedByController, err := func() (string, string, bool, error) {
+					lctx, lcancel := context.WithCancel(ctx)
+					defer lcancel()
```
Comment (MatousJobanek) [inline controllers/idler/idler_controller.go:146]: why is this needed?

Comment (filariow) [reply controllers/idler/idler_controller.go:146]: that's an error due to an improper use of `log.IntoContext`. let me refactor this

**Review by MatousJobanek — APPROVED** [review]
/lgtm
